### DISCUSSION TITLE
Soil Moisture (Capacitive & Battery) DIYMore 012592 - What GPIO is battery level?

### DISCUSSION THREAD

**Discussion by murfett-au** [arendst/Tasmota#19675] [Q&A]
This device I am discussing is: https://www.diymore.cc/collections/new-product/products/esp32-wifi-bluetooth-cp2104-dht11-soil-temperature-humidity-sensor-18650-battery-base

I have installed the template from https://templates.blakadder.com/diymore_012592.html on this device, and the moisture sensor signal is coming through on GPIO32, as the template suggests.

However, the GPIO34 connected to Analog 2 (ADC2) is always reading zero - both when the device is running on a battery, and when I power it from the USB (connected as required to flash Tasmota to it).

I tried configuring ADC3, ADC4 and ADC5 to read from GPIO35, GPIO36 and GPIO39, as these were the other 3 GPIOs marked with "IA" (for input analog) in the web ui "Configure Module" menu, but they also show around zero (they appear to be floating, as they move around 0,1,2,3,4 values).

I have tried tracing the circuit board from the + on the battery - there is no direct path from the battery + to any pin that I could find (not a suprise, as I would expect some voltage conditioning rather than a direct connection, but I looked anyway). Neither could I find a near zero ohm connection to anything on the PCB that looked like a three pin voltage regulator.

Has anyone worked out how to track the battery level on these devices?

**Comment by barbudor** [comment]
Just a basic question : If you are using ADC Range mode for Analog 2, what are the parameters for AdcParams ?
Bad range params could force a 0 result whatever is the value read frm the ADC
You could also change the mode for the ADC, from ADC Range to ADC Input

**Reply by murfett-au** [reply]
I have not changed these, so I assume it's the default mapping from 0->1023 to 0% to 100%. I will check.

**Reply by sfromis** [reply]
Do not assume that moisture sensing is anything like percentage humidity, and the numbers you get may not go from 0 to 1023 anyway.

**Reply by murfett-au** [reply]
the range is indeed the default:
```
00:04:31.460 CMD: ADCParam
00:04:31.471 MQT: stat/soilmoist2/RESULT = {"AdcParam1":[6,0,4095,0,100]}
00:04:44.861 CMD: Adcparam2
00:04:44.872 MQT: stat/soilmoist2/RESULT = {"AdcParam2":[6,0,4095,0,100]}
```

From the blakadder template page (https://templates.blakadder.com/diymore_012592.html):
"Note that the Soil humidity and battery level value are the raw value between 0 and 100% as tasmota does not support the capacitive soil sensor yet."

It "says on the side of the box" that this device measures battery level, but does not give a GPIO number, so my working hypothesis is that the battery level is measured on one of the GPIOs, but I don't know which one.

**Reply by murfett-au** [reply]
And thanks @sfromis , I am aware the raw value is an arbitrary indication of level of moisture derived from the capactiance of the soil around the sensor. I am calibrating that raw value to a point where I should start watering, and working out how long to run the water into the garden bed for. That's where the fun challenge lies.

That said, I am asking about the battery % - it is varying only between 0% and 1% now.

**Reply by sfromis** [reply]
The board may be based on https://github.com/pesor/TTGO-T-HIGrow which has `#define BAT_ADC 33`

**Comment by bjoernle** [comment]
When does tasmota support the soil moisture sensor?

**Reply by sfromis** [reply]
That is not a key question, as Tasmota does not need specific support for something delivering an analog signal. If things are set up properly, the ESP32 should be able to give an useful number about the sensed voltage between 0 and the max allowed 3.3V. It makes no difference whether that happens to be a stick you put into moist soil, or related to a battery.

It is up to you to make sure to use the correct wiring, GPIO and useful config for analog. Note that different ESP32 series have the *possible* analog pins on different GPIO ranges, and the Tasmota config process still includes ADC selections in the GPIO dropdown for pins not able to support ADC.

And of course, it happens that the producer of the board is not providing much useful info about how things are connected....

**Comment by msanaullahsahar** [comment]
@murfett-au Hi did you find any solution?

**Reply by murfett-au** [reply]
no
